test/provider_os_memory_not_impl.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2024 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "base.hpp"
+
+#include <umf/providers/provider_os_memory.h>
+
+using umf_test::test;
+
+TEST_F(test, os_provider_not_implemented) {
```

Comment:
you could also call `umfOsMemoryProviderOps` and check for NULL

Reply:
Done. Good catch.